src/inference/tests/unit/compiled_model_test.cpp

```diff
@@ -185,3 +185,39 @@ TEST_F(CompiledModelBaseTests, canReportErrorInExport) {
     EXPECT_CALL(*mock_compiled_model.get(), export_model(_)).WillOnce(Throw(std::runtime_error("compare")));
     OV_EXPECT_THROW_HAS_SUBSTRING(compiled_model.export_model(out_model), std::runtime_error, "compare");
 }
+
+class CompiledModelZeroBatch : public ::testing::Test {
```

Comment:
Could we also add some test cases for `model->reshape()` with `BS=0`

Reply:
Yeah, sure, will do. 
I have a question though. There are tests that use models with peculiar shapes and specify no layouts for them e.g. EyeTransformationTestsP.eye_decompose or ReferencePadV12TestNonConstPadsBeginPadsEndPadVal. 
I guess my initial assumption that we can expect only the BS on zero position was kinda stupid. I'm going to change the behavior to check for BS=0 only if the layout is specified, otherwise we cannot make any assumptions and thus cannot make the check valid